### PR TITLE
A small fix to put the full screen on the current monitor

### PR DESCRIPTION
**Pull request by stigbd**
I added one line to set the location of the new fullscreen. Tested in both one and two monitor settings. And it seamed to work fine, at least on my linux computer.
#229

### CODE REVIEW

**Comment by Phillipus** [comment]
Hi Stig, many thanks for taking a look at this. It seems strange that the Shell would be correct position when setting it to this:

fNewShell.setLocation(Display.getCurrent().getCursorLocation());

I wonder if this is the better way?

http://www.java2s.com/Tutorial/Java/0280__SWT/Centerashellontheprimarymonitor.htm

**Comment by stigbd** [comment]
Thanks for the reference. Yes, you are right. I reread the docs and found that my fix would put the full screen on whatever monitor your mouse pointer is on. (As such it can be considered a very tiny improvement ;-) )

The challenge is to identify the bounds for the monitor Archi is currently displayed on. The following piece of code seems to behave in the expected way, based on the link you provided.
```
        // To put the full screen on the current monitor:
        Rectangle bounds = fOldParent.getShell().getBounds();
        Rectangle rect = fNewShell.getBounds();
        int x = bounds.x + (bounds.width - rect.width) / 2;
        int y = bounds.y + (bounds.height - rect.height) / 2;
        fNewShell.setLocation(x, y);
```
I have updated my pull request with this new code.

File: com.archimatetool.editor/src/com/archimatetool/editor/diagram/actions/FullScreenAction.java [hunk 1]
```diff
@@ -25,9 +25,11 @@
 import org.eclipse.swt.events.KeyListener;
 import org.eclipse.swt.events.ShellAdapter;
 import org.eclipse.swt.events.ShellEvent;
+import org.eclipse.swt.graphics.Rectangle;
 import org.eclipse.swt.layout.FillLayout;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Monitor;
```
Comment (stigbd) [inline com.archimatetool.editor/src/com/archimatetool/editor/diagram/actions/FullScreenAction.java]: Sorry about this. Monitor is unused and should be removed.

**Comment by Phillipus** [comment]
Do you think, then, that your first fix was sufficient? Do you think that because the shell is full screen that it makes any difference to fNewShell.setLocation(x, y); ?

**Comment by stigbd** [comment]
No, I found a better way. My first fix "worked" because of to things: 1) I happened to put my mouse pointer on Archi, and then press F11, and 2) shell is full screen. When I put the mouse pointer on the other monitor and pressed F11, full screen would show up on other monitor.

My updated pull request should work in the intended way, no matter where the mouse pointer is. The  new code first identifies the bounds of the shell Archi is running in:
```
Rectangle bounds = fOldParent.getShell().getBounds();
```
This way, the new shell will be displayed on whatever monitor Archi is currently displaying on. It is, however, critical, that this code is executed before the call to full screen.

**Comment by Phillipus** [comment]
OK, do you think it would be possible to create just one commit from this and open a new PR? Just to keep the repo tidy? If it's a pain, I'll take it from here. Phil

**Comment by stigbd** [comment]
Created new, clean pull request that replaces this one. Thanks for useful comments! #231

**Pull request closed without merging**